src/AutoRest.CSharp.V3/ClientModels/ClientBuilder.cs

```diff
@@ -203,12 +225,19 @@ private static ServiceClientParameter BuildParameter(Parameter requestParameter)
                 defaultValue = ClientModelBuilderHelpers.ParseClientConstant(constantSchema);
             }
 
+            ParameterLocation? location = null;
+            if (requestParameter.Protocol.Http is HttpParameter httpParameter)
+            {
+                location = httpParameter.In;
+            }
+
             return new ServiceClientParameter(
                 requestParameter.CSharpName(),
                 CreateDescription(requestParameter),
                 ClientModelBuilderHelpers.CreateType(requestParameter.Schema, requestParameter.IsNullable()),
                 CreateDefaultValueConstant(requestParameter) ?? defaultValue,
-                requestParameter.Required == true);
+                requestParameter.Required == true,
+                location);
```

Comment:
Why do we need the parameter location?

Reply:
The location of the parameter determines which parameters are in which methods. See `IsNextPageParameter` in ClientWriter.

Reply:
I think building 2 instances of ClientMethod would help with this.

Reply:
Fixed.